src/DurableSDK/PowerShellServices.cs

```diff
@@ -6,50 +6,146 @@
+                }
+                hasExternalDurableSDK = false;
+            }
+
+            if (hasExternalDurableSDK)
+            {
+                SetFunctionInvocationContextCommand = $"{ExternalDurableSDKName}\\Set-FunctionInvocationContext";
+            }
+            else
+            {
+                SetFunctionInvocationContextCommand = $"{InternalDurableSDKName}\\Set-FunctionInvocationContext";
+            }
             _pwsh = pwsh;
         }
 
+        public bool HasExternalDurableSDK()
+        {
+            return hasExternalDurableSDK;
+        }
+
+        public PowerShell GetPowerShell()
+        {
+            return this._pwsh;
+        }
+
         public void SetDurableClient(object durableClient)
         {
             _pwsh.AddCommand(SetFunctionInvocationContextCommand)
                 .AddParameter("DurableClient", durableClient)
                 .InvokeAndClearCommands();
-
-            _hasSetOrchestrationContext = true;
+            hasInitializedDurableFunctions = true;
         }
 
-        public void SetOrchestrationContext(OrchestrationContext orchestrationContext)
+        public OrchestrationBindingInfo SetOrchestrationContext(
+            ParameterBinding context,
+            out IExternalOrchestrationInvoker externalInvoker)
         {
-            _pwsh.AddCommand(SetFunctionInvocationContextCommand)
-                .AddParameter("OrchestrationContext", orchestrationContext)
-                .InvokeAndClearCommands();
+            externalInvoker = null;
+            OrchestrationBindingInfo orchestrationBindingInfo = new OrchestrationBindingInfo(
+                context.Name,
+                JsonConvert.DeserializeObject<OrchestrationContext>(context.Data.String));
+
+            if (hasExternalDurableSDK)
+            {
+                Collection<Func<PowerShell, object>> output = _pwsh.AddCommand(SetFunctionInvocationContextCommand)
+                    // The external SetFunctionInvocationContextCommand expects a .json string to deserialize
+                    // and writes an invoker function to the output pipeline.
+                    .AddParameter("OrchestrationContext", context.Data.String)
+                    .InvokeAndClearCommands<Func<PowerShell, object>>();
+
+                // If more than 1 element is present in the output pipeline, we cannot trust that we have
+                // obtained the external orchestrator invoker; i.e the output contract is not met.
+                var outputContractIsMet = output.Count() == 1;
+                if (outputContractIsMet)
+                {
+                    externalInvoker = new ExternalInvoker(output[0]);
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Only a single output was expected for an invocation of {SetFunctionInvocationContextCommand}");
```

Comment:
Please add string resources for all messages that are displayed to the user.

Reply:
Added!